### PR TITLE
Use primary keys info if provided to build incremental query

### PR DESCRIPTION
**Pull request by ypc-faros**
## Description

Use primary keys info if provided to build incremental query

Note: es-lint forced me to rearrange the order of params in `createIncrementalQueriesV1/V2`. I think it should be fine since we don't have many users of the library (and less so of these functions) at this moment.

## Type of change
- [ ] Bug fix
- [X] New feature
- [X] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: src/graphql/graphql.ts [hunk 1]
```diff
@@ -1057,13 +1065,21 @@ export function createIncrementalReadersV1(
 
 export function createIncrementalQueriesV1(
   graphQLSchema: gql.GraphQLSchema,
-  avoidCollisions = true
+  avoidCollisions = true,
+  primaryKeys: Dictionary<ReadonlyArray<string>> | undefined = undefined
```
Comment (tovbinm) [inline src/graphql/graphql.ts:1069]: Remove undefined and just do "primaryKeys?"

File: src/graphql/graphql.ts [hunk 2]
```diff
@@ -1667,3 +1696,62 @@ function buildRefreshedFilter(
...
+        const type = this.graphQLSchema.getType(name);
+        if (this.isTopLevelModelTypeChecker(type)) {
+          result[name] = this.resolvePrimaryKey(type);
+        }
+      }
+
+      return result;
+  }
+
+  @Memoize()
```
Comment (tovbinm) [inline src/graphql/graphql.ts:1729]: Add type to memorize key

Comment (ypc-faros) [reply src/graphql/graphql.ts:1733]: From the [docs](https://www.npmjs.com/package/typescript-memoize) I believe it's not needed for single param methods?

Comment (tovbinm) [reply src/graphql/graphql.ts:1733]: Since your parameter is not a simple one, I would add a custom hash function

Comment (ypc-faros) [reply src/graphql/graphql.ts]: Done, thanks!

**Pull request merged**